packages/utils/test/ErrorSchemaBuilder.test.ts

```diff
@@ -0,0 +1,293 @@
+import { ErrorSchemaBuilder, ERRORS_KEY, ErrorSchema } from "../src";
+
+const AN_ERROR = "an error";
+const SOME_ERRORS = ["error1", "error2"];
+const STRING_PATH = "foo";
+const ARRAY_PATH = ["bar", "baz"];
+
+const INITIAL_ROOT = "initial root";
+const INITIAL_STRING = "initial string path";
+const INITIAL_ARRAY = "initial array path";
+
+// We have to cast to ErrorSchema because the type is not meant for building
+const INITIAL_SCHEMA: ErrorSchema = {
+  [ERRORS_KEY]: [INITIAL_ROOT],
+  [STRING_PATH]: {
+    [ERRORS_KEY]: [INITIAL_STRING],
+  } as ErrorSchema,
+  [ARRAY_PATH[0]]: {
+    [ARRAY_PATH[1]]: {
+      [ERRORS_KEY]: [INITIAL_ARRAY],
+    },
+  },
+} as ErrorSchema;
+
+describe("ErrorSchemaBuilder", () => {
+  describe("no initial schema", () => {
+    let builder: ErrorSchemaBuilder;
+    beforeAll(() => {
+      builder = new ErrorSchemaBuilder();
+    });
+    it("returns an empty error schema after construction", () => {
+      expect(builder.ErrorSchema).toEqual({});
+    });
+    it("adding error string without a path puts it into the root", () => {
+      expect(builder.addErrors(AN_ERROR).ErrorSchema).toEqual({
+        [ERRORS_KEY]: [AN_ERROR],
+      });
+    });
+    it("setting error string list without a path replaces at the root", () => {
+      expect(builder.setErrors(SOME_ERRORS).ErrorSchema).toEqual({
```

Comment:
I am personally not a fan of relying on the order of the tests to validate behavior, and would rather we start from scratch in `beforeEach`

Reply:
Given the differences in behavior of the `addErrors()` and `setErrors()` it seemed easier to build the tests this way. Did you want me to change it?

Reply:
It's fine, I agree this is easier and less code for something simple like this builder. Just a feeler to see how strongly you might agree 🙂

Reply:
In general I'm ok with doing incremental testing like this when it makes sense. Your feeler has revealed I weakly agree with you :)